### ISSUE TITLE
ESTK will fade away

### ISSUE THREAD

**Issue by fabianmoronzirfas** [basiljs/basil.js#205]
David Barranca pointed the following out in the Adobe Devs Slack channel:  

>Eric Robinson (not sure he's here as well) in the CC-dev PR pointed out that macOS High Sierra is going to be the latest OS to support 32bit applications. In other words, ESTK will keep working for another year or so. The clock is ticking!  

The response is:  

>Jonathan Ferman [7:10 PM] 
>@channel we are looking into ESTK and whether we update it for 64 bit, use another tool or do something else. Nothing to share at this point other than we are aware of the time ticking away.  

just so you know.  

(BTW as soon as junior is in the Kita I hope I'll find some time to get back onto the docs)

**Comment by trych** [comment]
Thanks for the heads-up! We'll see what happens. Meanwhile in my class, I will use the all new release of Sublime Text 3 after surviving the 4 or 5-year beta period. 😎

**Comment by ffd8** [comment]
Reopening this issue now for further discussion as Catalina (killing 32bit apps) has arrived and I slowly start to hear from former students who did the update (whoops) and ESTK died. Dug into some forums of rather pissed off individuals who depend on ESTK for their daily design business and are being told "don't upgrade". Curious if the community has enough pressure to encourage Adobe to upgrade it to 64bit or if their VS plugin will quickly evolve to a point of really replacing it? Luckily my last workshop with basil.js was on October 4th, just days before Catalina came out, so there wasn't an issue – the ESTK in the background with Sublime Text as coding environment was golden. But it's definitely going to be an issue in the upcoming months as more people update... *finally opens VS Code and downloads extension*

**Comment by fabianmoronzirfas** [comment]
IMO the Vscode plugin is fine for running scripts. It has not all the capabilites of ESTK but you run and debug scripts in it.

> One has to hit F5 to start the sketch

Never used this. I use the little palette of the debugger. In that dropdown you can set your default debugger I think. I never encountered you other issues.

**Comment by ffd8** [comment]
So in further VSCode exploration, finding a few good and strange things.
Based on [this guide](https://blob.pureandapplied.com.au/debugging-extendscript-files-with-vscode/), found a nice way to set current open file as the 'program' to run, by using the following for the launch.json config:

```
        {
            "type": "extendscript-debug",
            "request": "launch",
            "name": "Current file",
            "program": "${file}",
            "stopOnEntry": false
        }
```

And used on the following script (we should probably add the app target to our template, otherwise every time re-opening VSCode one would have to set it):

```
// @target indesign
// @includepath ~/Documents/;%USERPROFILE%Documents; 
// @include basiljs/basil.js; 
 
function draw() { 
    alert('hello')    
} 
```

However, each time I run a script, I get this annoying popup when done back in VSCode, saying "(#15)Execution halted":

<img width="424" alt="Screen Shot 2020-02-04 at 17 25 17" src="https://user-images.githubusercontent.com/570957/73802735-620c3600-4773-11ea-940f-782006f4c92f.PNG">

Any ideas what might be causing that??

**Comment by trych** [comment]
This is not really gonna solve your problem, but …

I gave up on the VS Script and the ExtendScript Debugger for now.
This tool is completely ridiculous and I have no idea, why Adobe thinks this is a proper replacement for ESTK (even though ESTK was not the ideal tool to begin with). Whenever I try to use it, it's somehow complicated to set up, behaves unpreditably and it is generally cumbersome and slow. Adobe seems to have *one* developer on it and according to some forum post she is on maternity leave until mid July (now don't get me wrong, I am all for people taking maternity leaves, but I think Adobe should have a few more developers on this. Have they never heard of the bus factor?).

I am still using Sublime Text and the Run in InDesign package. This has a few quirks as well, but I am currently getting up to speed with Sublime package development and am working on an improved version of the package (the original developer seems to have abandoned this tool). I will let you know as soon, as I have something usable. If you have specific requirements, I would also be happy to hear about them.

**Comment by fabianmoronzirfas** [comment]
@trych 🚀 I'm with you on this one. Tried to do a little script debugging for one of my students and it was painful. Did also the fallback to Sublime Text

**Comment by trych** [comment]
@fabianmoronzirfas Haha, I thought you were the last one to abandon VS Code. But that kind of confirms my impression. With the Sublime Text plugin my plans are as follows:

- generally tidying up
- allowing non ASCII characters to work in the console output (this also allows us to use proper pipe characters in the tree diagram of the `inspect()` console; got this to work alreay, was a super easy fix)
- allowing the use of the `// @targetengine` syntax
- etc.

**Comment by fabianmoronzirfas** [comment]
@trych 

> I thought you were the last one to abandon VS Code.

Only for Extendscript ;)

**Comment by ffd8** [comment]
I totally prefer the Sublime route, especially as I consider how multi-step and odd VSCode will be for completely beginners – nevertheless, this is all spawneed by the fact that ExtendScript ToolKit is now poof in 10.15 (32-bit with no plans to update). Fine for those with a running older system.. but slowly students are updating to the latest and ESTK won't open. Haha sooo I hope to make that switch as least painful as possible. Right now that popup on any script run isn't making it to pain-free... I suppose Windows users have no issues, but in our school (and most other design schools) it's 95% mac.

**Comment by trych** [comment]
I would just forward everybody to Sublime for now (if they want to work with basil.js or ES). I think it's a fantastic piece of software.

And yes, I know it's not free, but the fact that you can evaluate it for free indefinitely makes it work really well in seminars and such. And it even offers a portable version so that students won't run into admin rights/installation issues.

**Comment by ffd8** [comment]
Ahhh I had just written a whole long blurb further researching workarounds for 10.15+ future without ESTK, since my workflow in Sublime w/ Run In InDesign always required ESTK running in the background, only to realize (now to actually reference the old #15 ) that it was simply because the `clearConsole()` function forces ESTK to open?! Anyone know if this function is still necessary?? The console in SublimeText is working perfect for println/inspect/etc – and quick test still has ESTK working without it... soooo.. I think this should either be removed or use a flag to disable? Looking forward to juust Sublime Text + basil.js w/ students!

@trych had you disabled this too or were you also using ESTK open in the background?

**Comment by fabianmoronzirfas** [comment]
We should remove everything that could trigger the ESTK by default.

**Comment by trych** [comment]
Fixed the `clearConsole()` thing in #343. Will close this issue here for now. Feel free to re-open if anything related comes up again.
